kernel/signal.c

```diff
@@ -820,17 +839,32 @@ int myst_signal_altstack(const stack_t* ss, stack_t* old_ss)
         if (me->signal.altstack.ss_flags & SS_ONSTACK)
             ERAISE(-EPERM);
 
+        myst_process_t* process = myst_process_self();
+
         if (ss->ss_flags & SS_DISABLE)
         {
             me->signal.altstack.ss_flags |= SS_DISABLE;
             me->signal.altstack.ss_sp = NULL;
             me->signal.altstack.ss_size = 0;
+
+            /* clear the signal delivery altstack */
+            ECHECK(myst_set_signal_delivery_altstack(
```

Comment:
The function call doesn't seem to match the comment.

Reply:
Fixed.